### PR TITLE
Add option to pass rotations through Jabalizer.

### PR DESCRIPTION
**Pull request by mstechly**
## Description

Added option to pass rotations through Jabalizer – this means that if one uses a circuit with rotations (RX, RY or RZ), Jabalizer will treat them in the same way it usually treats "T" gate. 

Current version of this code is quite messy and untested, however, in order to do it properly a major refactor of our resource estimation code is needed, which is beyond the scope of this PR and should be done separately.

### CODE REVIEW

File: src/benchq/resource_estimation/graph_compilation_rotations.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+        * distance
+        * 6
+        * physical_gate_time
+    )
+
+
+def is_circuit_in_the_right_format(circuit: Circuit) -> bool:
+    for operation in circuit.operations:
+        # TODO: refactor to be less hacky
+        if operation.gate.name not in CLIFFORD_GATES + ["T", "RZ", "RX"]:
```
Comment (AthenaCaesura) [inline src/benchq/resource_estimation/graph_compilation_rotations.py:59]:
```suggestion
        if operation.gate.name not in CLIFFORD_GATES + ["T", "RZ", "RY", "RX"]:
```

**Review by AthenaCaesura — COMMENTED** [review]
Does this code work with `QuantumProgram`s? It really should.

**Comment by mstechly** [comment]
Yes it does, I just didn't add the extra example.

**Review by AthenaCaesura — APPROVED** [review]

File: src/benchq/compilation/jabalizer_wrapper.jl [hunk 2]
```diff
@@ -59,16 +59,17 @@ end
 
 function prepare(num_qubits, qubit_map, icm_output)
     gate_map = Dict("I" => Jabalizer.Id,
```
[the remark this answers is not in the document]
Comment (mstechly) [reply src/benchq/compilation/jabalizer_wrapper.jl:61]: True, but black automatically removes the tabs for me :/

**Pull request merged**